gapis/resolve/memory.go

```diff
@@ -115,3 +124,91 @@ func Memory(ctx context.Context, p *path.Memory, rc *path.ResolveConfig) (*servi
 		Observed: service.NewMemoryRanges(observed),
 	}, nil
 }
+
+// MemoryAsType resolves and returns the memory from the path p.
+func MemoryAsType(ctx context.Context, p *path.MemoryAsType, rc *path.ResolveConfig) (*memory_box.Value, error) {
+	ctx = SetupContext(ctx, path.FindCapture(p), rc)
+
+	cmdIdx := p.After.Indices[0]
+	fullCmdIdx := p.After.Indices
+
+	allCmds, err := Cmds(ctx, path.FindCapture(p))
+	if err != nil {
+		return nil, err
+	}
+
+	if count := uint64(len(allCmds)); cmdIdx >= count {
+		return nil, errPathOOB(cmdIdx, "Index", 0, count-1, p)
+	}
+
+	sd, err := SyncData(ctx, path.FindCapture(p))
+	if err != nil {
+		return nil, err
+	}
+
+	cmds, err := sync.MutationCmdsFor(ctx, path.FindCapture(p), sd, allCmds, api.CmdID(cmdIdx), fullCmdIdx[1:], true)
+	if err != nil {
+		return nil, err
+	}
+
+	defer analytics.SendTiming("resolve", "memory")(analytics.Count(len(cmds)))
+
+	s, err := capture.NewState(ctx)
+	if err != nil {
+		return nil, err
+	}
+	err = api.ForeachCmd(ctx, cmds, func(ctx context.Context, id api.CmdID, cmd api.Cmd) error {
+		cmd.Mutate(ctx, id, s, nil, nil)
```

Comment:
do we need to check the result of the `Mutate` here?

Reply:
This was mimicing the above call in memory.go in which we do not check the return. Typically we don't expect mutate to fail in the case where we haven't run a transform.